addons/mail/static/src/js/thread.js

```diff
@@ -116,7 +116,7 @@ var Thread = Widget.extend({
 
         this.$('[data-o-mail-quote="1"]').each(function () {
             var $content = $(this);
-            var $read_more = $('<a class="o_mail_read_more" href="#"></a>').text(read_more);
+            var $read_more = $('<a class="o_mail_read_more" />').text(read_more);
             var is_read_more = true;
             $read_more.click(function() {
                 is_read_more = !is_read_more;
```

Comment:
This function takes event `e` as argument. `e.preventDefault()` would then do the trick.